built_value_generator/lib/src/serializer_source_field.dart

```diff
@@ -193,12 +193,24 @@ abstract class SerializerSourceField
       {bool topLevel = true}) {
     var bareType = _getBareType(type);
 
-    // For built collections we can cast to the bare type when deserializing,
-    // instead of the full generic type. This is because the `replace` method
+    // For built collections we can cast to the dynamic type when deserializing,
+    // instead of the actual generic type. This is because the `replace` method
     // checks the generic type and copies if needed.
-    var generics = topLevel && DartTypes.isBuiltCollectionTypeName(bareType)
-        ? ''
-        : _getGenerics(type);
+    String generics;
+    if (topLevel && DartTypes.isBuiltCollectionTypeName(bareType)) {
+      if (bareType == 'BuiltList' || bareType == 'BuiltSet') {
+        generics = 'dynamic';
+      } else if (bareType == 'BuiltMap' ||
+          bareType == 'BuiltListMultimap' ||
+          bareType == 'BuiltSetMultimap') {
+        generics = 'dynamic, dynamic';
+      } else {
+        // Should not get here; only 5 classes satisfy
+        // `isBuiltCollectionTypeName`.
```

Comment:
WDYT for here? throw? throw Unsupported or something?

Reply:
throw UnsupportedError SGTM.

Reply:
Done